src/main/java/com/pusher/rest/Pusher.java

```diff
@@ -306,7 +302,7 @@ public Result trigger(final List<String> channels, final String eventName, final
 
         final String body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
 
-        return post("/events", body);
+        return new TriggerResult(post("/events", body), dataMarshaller);
```

Comment:
The `dataMarshaller` is only intended for marshalling the client's data payload (for which we are format agnostic), allowing them to e.g. have their data payload delivered as msgpack or a string, or whatever.

Our responses are strictly JSON, so we should always use Gson to decode them.